pkg/kubelet/apis/nodeconfig/register.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodeconfig
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var (
+	SchemeBuilder = runtime.NewSchemeBuilder(addKnownTypes)
+	AddToScheme   = SchemeBuilder.AddToScheme
+)
+
+// GroupName is the group name use in this package
+const GroupName = "nodeconfig"
```

Comment:
The KubeletConfiguration (and decomposition) describe how you want your node to behave, but the Kubelet is really just an implementation detail of how these node-level things are configured, so it doesn't make sense to marry the name to the Kubelet (even though the Kubelet tree is, at present, the best location for the types - because the Kubelet is the only thing that implements them).

Those node-agents should expose their configuration types from their own trees. Interesting idea about putting them in the same group though. Can you get name collisions between components' API groups during registration (like if kube-proxy also called its API group `nodeconfig` and had some dependency on the Kubelet)? Come to think of it this seems possible... maybe it would be better to call the group `kubeletconfig` just to be safe... This might also be less confusing for people.

My bias is toward trying to design higher-level APIs, and "configure general node behavior" is a higher level concept than "configure the Kubelet". I think it also gets people to think in terms of what's really necessary for achieving the node behavior they want, rather than putting all kinds of unnecessary stuff in the configuration API.

WDYT?